utils/wasm-gen/src/wasm.rs

```diff
@@ -197,46 +197,84 @@ impl PageCount {
     }
 }
 
-/// Represents memory layout that can be safely used between syscalls and
-/// instructions.
-///
-/// The last memory page in program generated by `wasm-gen` is reserved for
-/// internal use. Currently, we take [`MemoryLayout::RESERVED_MEMORY_SIZE`]
-/// bytes from the last memory page and also prohibit modification of this
-/// memory at the `wasm-smith` and `wasm-gen` level.
-///
-/// If you want to store some data in memory and then access it in the program,
-/// consider adding a new pointer to this structure.
-pub struct MemoryLayout {
-    pub init_called_ptr: i32,
-    pub wait_called_ptr: i32,
-    pub remaining_memory_len: u32,
-    pub remaining_memory_ptr: i32,
+macro_rules! def_memory_layout {
```

Comment:
whats the point of macro?

Reply:
This structure defines what the reserved memory will look like. Macro is needed to calculate offsets.